src/main/java/io/github/thebusybiscuit/slimefun4/utils/RechargeableHelper.java

```diff
@@ -69,12 +77,14 @@ static float getCharge(@Nonnull ItemMeta meta) {
             for (String line : meta.getLore()) {
                 if (REGEX.matcher(line).matches()) {
                     String data = ChatColor.stripColor(PatternUtils.SLASH_SEPARATOR.split(line)[0].replace(LORE_PREFIX, ""));
-                    return Float.parseFloat(data);
+
+                    float loreValue = Float.parseFloat(data);
+                    meta.getPersistentDataContainer().set(key, PersistentDataType.FLOAT, loreValue);
```

Comment:
`meta.getPersistentDataContainer()` is called twice here, you could have it as a variable further up.